File: src/lib/shared/types/Artist.ts
import type {ExternalUrls} from "./ExternalUrls";
import type {Followers} from "./Followers";
import type {Image} from "./Image";

export interface Artist {
  external_urls?: ExternalUrls
  followers?: Followers
  genres?: string[]
  href: string
  id: string
  images?: Image[]
  name: string
  popularity?: number
  type?: string
  uri?: string
}

File: lib/api/schemas/artist.ex
defmodule Artist do
  use Api.Schema

  schema "artists" do
    embeds_one :external_urls, ExternalUrls
    embeds_one :followers, Followers
    field :genres, {:array, :string}
    field :href, :string
    field :spotify_id, :string
    embeds_many :images, Image
    field :name, :string
    field :popularity, :integer
    field :type, Ecto.Enum, values: [:artist]
    field :uri, :string
  end
end
